chainstate/test-suite/src/tests/helpers/pos.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate_storage::Transactional;
+use chainstate_test_framework::TestFramework;
+use chainstate_types::{pos_randomness::PoSRandomness, vrf_tools::construct_transcript};
+use common::{
+    chain::{
+        block::{consensus_data::PoSData, timestamp::BlockTimestamp},
+        config::EpochIndex,
+        signature::inputsig::InputWitness,
+        OutPoint, PoolId, RequiredConsensus,
+    },
+    primitives::{Amount, BlockHeight, Compact},
+};
+use consensus::ConsensusPoSError;
+use crypto::vrf::{VRFPrivateKey, VRFPublicKey};
+
+use super::block_index_handle_impl::TestBlockIndexHandle;
+
+#[allow(clippy::too_many_arguments)]
+pub fn pos_mine(
+    initial_timestamp: BlockTimestamp,
+    kernel_outpoint: OutPoint,
+    vrf_pk: &VRFPublicKey,
```

Comment:
Nitpick: You can produce the public key from the private key. 

Maybe this is irrelevant if this is part of the test where an invalid case is tested as well.